rai/blockstore.cpp

```diff
@@ -838,23 +873,31 @@ bool rai::block_store::account_exists (MDB_txn * transaction_a, rai::account con
 	return iterator != rai::store_iterator (nullptr) && rai::account (iterator->first.uint256 ()) == account_a;
 }
 
-bool rai::block_store::account_get (MDB_txn * transaction_a, rai::account const & account_a, rai::account_info & info_a)
+auto rai::block_store::account_get (MDB_txn * transaction_a, rai::account const & account_a) -> expected<rai::account_info, std::error_code>
 {
+	expected<rai::account_info, std::error_code> res;
+	rai::account_info info_a;
 	rai::mdb_val value;
 	auto status (mdb_get (transaction_a, accounts, rai::mdb_val (account_a), value));
 	assert (status == 0 || status == MDB_NOTFOUND);
-	bool result;
 	if (status == MDB_NOTFOUND)
 	{
-		result = true;
+		res = make_unexpected (make_error_code (error::missing_account));
```

Comment:
This idiom is what people seem to use, but I think better ergonomics should be possible. For a class using just enums (instead of system::error_code), this would be `make_unexpected (error::missing_account);`